### PR TITLE
masterdata manage translations

### PR DESCRIPTION
**Pull request by dageorgero**

### CODE REVIEW

**Review by meghuizen — APPROVED** [review]

File: Core/Data.EF/ApplicationDbContext.cs [hunk 1]
```diff
@@ -94,6 +97,7 @@ public override int SaveChanges(bool acceptAllChangesOnSuccess)
     public override async Task<int> SaveChangesAsync(bool acceptAllChangesOnSuccess, CancellationToken cancellationToken = default)
     {
         SetCreatedUpdatedFields();
+        InitializeTranslation();
```
Comment (Johanh-od) [inline Core/Data.EF/ApplicationDbContext.cs:100]: Do we really want this in the ApplicationDbContext and forcing every context to use this? @jstorteboom

**Review by Johanh-od — CHANGES_REQUESTED** [review]

**Comment by Johanh-od** [comment]
Can we add a short readme how you initialize and configure the translations?

File: Services/Masterdata/Services.Masterdata/Helpers/LanguageHelper.cs [hunk 2]
```diff
@@ -0,0 +1,571 @@
+﻿namespace Lens.Services.Masterdata.Helpers;
+public static class LanguageHelper
```
Comment (Johanh-od) [inline Services/Masterdata/Services.Masterdata/Helpers/LanguageHelper.cs:2]: Lets rephrase this to LanguageConfiguration and move it to a Core lib, so multi language support is available from the core and not masterdata specific

File: Services/Masterdata/Services.Masterdata/Repositories/IMasterdataRepository.cs [hunk 3]
```diff
@@ -1,60 +1,62 @@
 ﻿using Lens.Core.Lib.Models;
+using Lens.Core.Data.EF.Translation.Models;
 using Lens.Services.Masterdata.Models;
 
 namespace Lens.Services.Masterdata.Repositories;
 
 public interface IMasterdataRepository
```
Comment (Johanh-od) [inline Services/Masterdata/Services.Masterdata/Repositories/IMasterdataRepository.cs:7]: Tip: you better split this repo **in the future**, so it has only 1 responsibility instead of 2 (masterdata type & master data item logic)

Comment (Johanh-od) [reply Services/Masterdata/Services.Masterdata/Repositories/IMasterdataRepository.cs:7]: It also reduces the lines of code in 1 file if the class has only 1 responsibility

Comment (Johanh-od) [reply Services/Masterdata/Services.Masterdata/Repositories/IMasterdataRepository.cs:7]: It also prevents the use of the (ugly) region statements in your file :)

File: Services/Masterdata/Services.Masterdata/Helpers/LanguageHelper.cs (hunk 2 above)
Comment (dageorgero) [reply Services/Masterdata/Services.Masterdata/Helpers/LanguageHelper.cs:2]: i want to keep it on a masterdata level for now. the translation is functional for now on a masterdata level. we can move this language configuration file in a future story, when the translation will be fully developed for on a framework level!

**Comment by dageorgero** [comment]
> Can we add a short readme how you initialize and configure the translations?
I think for now it should be used/tested on masterdata. It needs some fine tuning to be more developer friendly for using in other apps.
